plutus-tx-plugin/src/PlutusTx/Compiler/Type.hs

```diff
@@ -146,7 +146,9 @@ compileTyCon tc
                     PIR.defineDatatype @_ @uni (LexName tcName) (PIR.Def tvd fakeDatatype) Set.empty
 
                     -- Type variables are in scope for the rest of the definition
-                    withTyVarsScoped (GHC.tyConTyVars tc) $ \tvs -> do
+                    -- We remove vars of 'RuntimeRep' type to properly compile '(#, #)' type constructor
+                    let filteredTyVars = dropWhile (GHC.isRuntimeRepTy . GHC.varType) $ GHC.tyConTyVars tc
```

Comment:
To avoid such a general behaviour we can case on `tycon` to process `(#, #)` in a individual order.

Same thing is possible in all places where we drop `RuntimeRep` args I guess...

Reply:
I think I'm okay with that so long as we document it. I don't think there will be that many other things with `RuntimeRep` arguments, and I'm okay if they behave slightly weirdly.

Reply:
We could try including an example of a custom rep-polymorphic datatype and seeing what happens! I don't know how to write one, but we can probably figure it out!